src/background/actions/getSigningCertificate.ts

```diff
@@ -0,0 +1,91 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import Action from "@web-eid.js/models/Action";
+import { NativeGetSigningCertificateRequest } from "@web-eid.js/models/message/NativeRequest";
+import { NativeGetSigningCertificateResponse } from "@web-eid.js/models/message/NativeResponse";
+import UnknownError from "@web-eid.js/errors/UnknownError";
+import UserTimeoutError from "@web-eid.js/errors/UserTimeoutError";
+import { serializeError } from "@web-eid.js/utils/errorSerializer";
+
+import {
+  ExtensionFailureResponse,
+  ExtensionGetSigningCertificateResponse,
+} from "@web-eid.js/models/message/ExtensionResponse";
+
+import { MessageSender } from "../../models/Browser/Runtime";
+import NativeAppService from "../services/NativeAppService";
+import config from "../../config";
+import { getSenderUrl } from "../../shared/utils/sender";
+import { throwAfterTimeout } from "../../shared/utils/timing";
+
+export default async function getSigningCertificate(
+  sender: MessageSender,
+  userInteractionTimeout: number,
+  lang?: string,
+): Promise<ExtensionGetSigningCertificateResponse | ExtensionFailureResponse> {
+  const extension        = config.VERSION;
+  const nativeAppService = new NativeAppService();
+
+  try {
+    const nativeAppStatus = await nativeAppService.connect();
+
+    config.DEBUG && console.log("getSigningCertificate: connected to native", nativeAppStatus);
+
+    const message: NativeGetSigningCertificateRequest = {
+      command: "get-signing-certificate",
+
+      arguments: {
+        origin: (new URL(getSenderUrl(sender))).origin,
+
+        ...(lang ? { lang } : {}),
+      },
+    };
+
+    const response = await Promise.race([
+      nativeAppService.send<NativeGetSigningCertificateResponse>(message),
+      throwAfterTimeout(userInteractionTimeout, new UserTimeoutError()),
+    ]);
+
+    const isResponseValid = (
+      response?.certificate &&
+      response?.supportedSignatureAlgorithms.length
+    );
+
+    if (isResponseValid) {
+      return { action: Action.GET_SIGNING_CERTIFICATE_SUCCESS, ...response };
+    } else {
+      throw new UnknownError("unexpected response from native application");
+    }
+  } catch (error: any) {
+    error.extension = extension;
```

Comment:
Ditto, `error.extensionVersion`?

Also, I see that this is inconsistent with other commands - I presume you intend to add extension version to other command's errors as well?

Reply:
Removed.
The next line was also wrong, fixed it.